src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -37,19 +45,24 @@ def create_configstore(client,
                        enable_public_network=None,
                        disable_local_auth=None,
                        retention_days=None,
-                       enable_purge_protection=None):
+                       enable_purge_protection=None,
+                       no_replica=False,
+                       replica_name=None,
+                       replica_location=None):
     if assign_identity is not None and not assign_identity:
         assign_identity = [SYSTEM_ASSIGNED_IDENTITY]
 
     public_network_access = None
     if enable_public_network is not None:
         public_network_access = 'Enabled' if enable_public_network else 'Disabled'
 
-    if sku.lower() == 'free' and (enable_purge_protection or retention_days):
-        logger.warning("Options '--enable-purge-protection' and '--retention-days' will be ignored when creating a free store.")
+    if sku.lower() == 'free' and (enable_purge_protection or retention_days or replica_name or replica_location or no_replica):
+        logger.warning("Options '--enable-purge-protection', '--replica-name', '--replica-location' , 'no-replica' and '--retention-days' will be ignored when creating a free store.")
         retention_days = None
         enable_purge_protection = None
 
+    __validate_replication(sku, replica_name, replica_location, no_replica)
```

Comment:
I think we should move this validation to _validators.py

Reply:
We move the whole function to validators.py?

Reply:
yes

Reply:
@avanigupta I initially had a similar thought, but since this method jointly validates 3 arguments, I was wondering which of them the validation should be added to in params.

Reply:
I think it should be on the sku.

Reply:
The warning about free sku just above this code can also move to validators

Reply:
Sounds good. I have moved the function to validators.